src/ucs/memory/rcache.h

```diff
@@ -216,4 +220,21 @@ void ucs_rcache_region_hold(ucs_rcache_t *rcache, ucs_rcache_region_t *region);
 void ucs_rcache_region_put(ucs_rcache_t *rcache, ucs_rcache_region_t *region);
 
 
+/**
+  * Decrement memory region reference count, invalidate and possibly destroy it.
+  *
+  * @param [in] rcache    Memory registration cache.
+  * @param [in] region    Memory region to release.
+  * @param [in] cb        Completion callback. Can be NULL, which means that
+  *                       no completion will be generated in case of
+  *                       outstanding operation. Completion callback is called
+  *                       when region is released.
```

Comment:
from which context is the callback called? is there a lock held? can the callback call any function?

Reply:
it is called from context where md_dereg is called or rcache is released, no rcache logs are held

Reply:
update: callback is called from pg_lock held (else Coverity failed to handle such behavior)

Reply:
this means the callback cannot do any operations which may access the rcache? need to document it explicitly

Reply:
ok, added info into function description